packages/cdktf-cli/bin/cmds/helper/check-environment.ts

```diff
@@ -48,19 +49,97 @@ async function checkNodeVersion() {
   throwIfLowerVersion("Node.js", "14.17.0", out);
 }
 
-export async function checkEnvironment(projectPath = process.cwd()) {
-  await checkNodeVersion();
-
-  let language: string | undefined;
+function getLanguage(projectPath = process.cwd()): string | undefined {
   try {
     const cdktfJson = require(path.resolve(projectPath, "cdktf.json"));
-    language = cdktfJson.language;
+    return cdktfJson.language;
   } catch {
     // We can not detect the language
   }
+}
 
-  switch (language) {
+export async function checkEnvironment() {
+  await checkNodeVersion();
+
+  switch (getLanguage()) {
     case "go":
       await checkGoVersion();
   }
 }
+
+async function getNodeModuleVersion(): Promise<string> {
+  let output;
+  try {
+    output = await exec("npm", ["list", "cdktf", "--json"], {
+      env: process.env,
+    });
+  } catch (e) {
+    throw Errors.Usage(`Unable to run 'npm list cdktf --json': ${e}`);
+  }
+
+  let json;
+  try {
+    json = JSON.parse(output);
+  } catch (e) {
+    throw Errors.Usage(
+      `Unable to parse output of 'npm list cdktf --json': ${e}`
+    );
+  }
+
+  if (
+    !json.dependencies ||
+    !json.dependencies.cdktf ||
+    !json.dependencies.cdktf.version
+  ) {
+    throw Errors.Usage(
+      `Unable to find 'cdktf' in 'npm list cdktf --json': ${output}`
+    );
+  }
+
+  return json.dependencies.cdktf.version;
+}
+
+export async function verifySimilarLibraryVersion() {
+  const language = getLanguage();
+  if (!language) {
+    // We could not detect the language, disabling the check
+    return;
+  }
+
+  const noOp = async () => {}; // eslint-disable-line @typescript-eslint/no-empty-function
+  const getLibraryVersionMap: Record<
+    string,
+    () => Promise<string | undefined | void>
+  > = {
+    typescript: getNodeModuleVersion,
+    go: noOp,
+    python: noOp,
+    csharp: noOp,
+    java: noOp,
+  };
+
+  const libVersion = await (getLibraryVersionMap[language] || noOp)();
```

Comment:
If we narrow the type of the return value of `getLanguage`, we wouldn't need this `|| noOp`, right?

Reply:
Yes, but we would lie to TS since anyone can edit the cdktf.json :) we could use zod here to parse, but I felt like that might be a bit of an overkill

Reply:
Yeah, true!